### DISCUSSION TITLE
Support for NodeJS in roslibjs

### DISCUSSION THREAD

**Discussion by faltenberg** [foxglove/community#740] [Q&A]
Are there plans to bring RosLibNode.js and subsequent files back to the @foxglove/roslibjs library? The original roslib library by RobotWebTools has issues and incompabilities with ROS2. I want to migrate to your fork instead. However, support for NodeJS was dropped as it seems.

Edit: when running the ros-humble-foxglove-bridge-server and trying to use a simple roslibjs nodejs example I run into incompabilities.
```
[foxglove_bridge-1] [INFO] [1686475892.184255200] [foxglove_bridge]: [WS] Rejecting client 172.27.0.2:32932 which did not declare support for subprotocol foxglove.websocket.v1
Jun 11 11:31:32 gl-NAB5 bash[835]: [foxglove_bridge-1] [ERROR] [1686475892.184347798] [foxglove_bridge]: [WS] Handshake ended with HTTP error: 400
```

I have traversed your @foxglove/roslibjs library to check where it sets the websocket's subprotocol to "foxglove.websocket.v1" (as indicated in [foxglove/ws-protocol](https://github.com/foxglove/ws-protocol/blob/main/docs/spec.md). I could not find no such indicator in the @foxglove/roslibjs code base. So I wonder how it connects to the foxglove-rosbride-server, if it does already. I could not test the included examples to verify that the library does indeed work already with your rosbridge.

**Comment by defunctzombie** [comment]
Our roslibjs fork is used to power the [rosbridge](https://foxglove.dev/docs/studio/connection/ros2#rosbridge) connection for Foxglove Studio. We do not plan to add or maintain any features beyond what we need for Studio.

For connecting to a Foxglove websocket protocol server (like the one that powers the [ros-foxglove-bridge](https://github.com/foxglove/ros-foxglove-bridge) you can find examples in the ws-protocol repo. Here is one such example in typescript: https://github.com/foxglove/ws-protocol/blob/main/typescript/ws-protocol-examples/src/examples/simple-client.ts that should also work in node.js.
